plugin.go

```diff
@@ -158,19 +162,18 @@ func NewCompiledPlugin(
 	//  - If there is only one module in the manifest then that is the main module by default
 	//  - Otherwise the last module listed is the main module
 
-	modules := map[string]wazero.CompiledModule{}
 	for i, wasm := range manifest.Wasm {
 		data, err := wasm.ToWasmData(ctx)
 		if err != nil {
 			return nil, err
 		}
 
-		_, mainExists := modules["main"]
+		_, mainExists := p.modules["main"]
```

Comment:
`mainExists` will always be `false` since we never add `main` module to the map

Reply:
great catch!